lib/shopify-cli/context.rb

```diff
@@ -456,5 +480,21 @@ def ctx_path(fname)
         File.join(root, fname)
       end
     end
+
+    def retrieve_latest_gem_version
+      response = Net::HTTP.get_response(GEM_LATEST_URI)
+      latest = JSON.parse(response.body)
+      latest["version"]
+    rescue
+      nil
```

Comment:
Is the intent here that if there's a network error or JSON parse error, this will return nil? Is it kosher to be rescuing _any_ kind of exception here?

Reply:
W.r.t. printing it after rather than before, several commands e.g., `serve`, `create`, `populate` end with information and instructions, including links, that the user needs to be aware of (potentially even copy).  Having the version notice at the end could detract from the instructional information where it exists.

W.r.t. the rescue, given that this is mainly an informational notice about a new version, if anything goes wrong with the network request or JSON error at that particular time, I don't think it's a huge deal - just wait 24 hours and try again.

Reply:
I'm fine with the concept of rescuing from those specific errors and just punting for sure. My question was more about whether it's ok to rescue from _any_ exception here, as opposed to the specific errors we anticipate.

Reply:
Yeah, I think so in this case, since any error that occurs only impacts the potential display of an informational message, vs. impacting the actual utility of the CLI.